flow/workflows/maintenance_flow.go

```diff
@@ -72,19 +76,59 @@ func StartMaintenanceWorkflow(ctx workflow.Context, input *protos.StartMaintenan
 }
 
 func startMaintenance(ctx workflow.Context, logger log.Logger) (*protos.StartMaintenanceFlowOutput, error) {
+	signalChan := model.StartMaintenanceSignal.GetSignalChannel(ctx)
+	maintenanceSelector := workflow.NewNamedSelector(ctx, "MaintenanceLoop")
+	skippedFlows := make(map[string]struct{})
+	cancelCurrentChild := func() {}
+
+	signalChan.AddToSelector(maintenanceSelector, func(maintenanceSignal *protos.StartMaintenanceSignal, _ bool) {
+		logger.Info("Received StartMaintenance Signal", slog.Any("signal", maintenanceSignal))
+		newSkippedFlows := make(map[string]struct{})
+		maps.Copy(newSkippedFlows, skippedFlows)
+		for _, flow := range maintenanceSignal.SkippedSnapshotWaitFlows {
+			newSkippedFlows[flow] = struct{}{}
+		}
+		skippedFlows = newSkippedFlows
+		cancelCurrentChild()
+	})
+	workflow.Go(ctx, func(ctx workflow.Context) {
+		for {
+			maintenanceSelector.Select(ctx)
+		}
```

Comment:
need to break loop when context done. you'll also need to add  `ctx.Done()` to `maintenanceSelect`

Reply:
> I would suggest trying to do this without `workflow.Go`

Is that an antipattern? I am mainly trying to process signals in background an using cancellation of the activity as a downstream signalling mechanism

https://docs.temporal.io/develop/go/message-passing#signals